tests/core_tests/rct.cpp

```diff
@@ -57,7 +57,7 @@ bool gen_rct_tx_validation_base::generate_with(std::vector<test_event_entry>& ev
     miner_accounts[n].generate();
     CHECK_AND_ASSERT_MES(generator.construct_block_manually(blocks[n], *prev_block, miner_accounts[n],
         test_generator::bf_major_ver | test_generator::bf_minor_ver | test_generator::bf_timestamp | test_generator::bf_hf_version,
-        2, 2, prev_block->timestamp + DIFFICULTY_BLOCKS_ESTIMATE_TIMESPAN * 2, // v2 has blocks twice as long
+        2, 2, prev_block->timestamp + DIFFICULTY_BLOCKS_ESTIMATE_TIMESPAN * 4, // v2 has blocks four times as long
```

Comment:
That seems predicated on height though, not block version AFAICT, from your code above

Reply:
As in `get_block_reward` https://github.com/aeonix/aeon-rebase/pull/4/commits/bf320b8e504c48278e32ad2660b246cdeb10e921#diff-1a57d4e6013984c420da98d1adde0eafR91 , I made a little trick in the logic such that the block time becomes V2 either when 1) the network version is 1, or 2) the block height reaches `HARDFORK_1_HEIGHT `. Because we explicitly set the hf version to 2, the block time should be 240s. Do I make sense?

Reply:
Yes